postmortems/2021-07-27_update_ris3.md

```diff
@@ -0,0 +1,67 @@
+
+### Update attempt 3
+
+The storypoint was eventually created using the following incredibly brittle method:
+
+- Add the new storypoint into `ris3_views.py` and run this transform only to generate the triples for the storypoint.
+- Open the triples json and find and copy only the new triples.
+- Open the model pulled down from staging and paste the new triples into it, after making sure that `asOrdinal` wasn't going to collide with any existing values.
+- Also edit the markdown for the sidebar in the model to add a link to the new storypoint.
+- Push the model back up to staging.
+
+(This involved a lot of direct editing of a 2.5MB slab of JSON representing the model, which caused Visual Studio Code to crash twice.)
+
+Obviously I have compounded the original problem here by adding even more unversioned data to the model -- although I have made a PR with an update to `ris3_views`, the generated triples will have the wrong tileset version as there's no good way of setting this on a per-storypoint level yet. Even if there was, though, `ris3_views` is still out-of-sync with `ris3_points.ttl`.
+
+### A note on tile generation
+
+Tile generation from PDFs is usually done On Martin's Computer using a shell script. These shell scripts are undocumented; Martin provided one for RIS3 prior to going on holiday but it started with a huge array of commented-out variables (presumably corresponding to the different tilesets for RIS3) that made things generally rather impenetrable. Now that I have gone through this learning process I *think* I could probably use this to generate some tiles, but when looking at it from the other end it was much easier to let Mark N do it as he knew the AI-native method would work.
+
+## Actions
+
+### Immediate (now)
+
+- **Locate the copy of `ris3_points.ttl` was used to generate the existing model on staging and prod and version it.** All code that is used to generate production data should be versioned so that the production data can be reconstructed from scratch if need be.
+
+### Near future (2-4 weeks)
+
+- Backport the manual data edits from `ris3_points.ttl` into `ris3_views.py`.
+- Give `ris3_views.py` the ability to set specific tileset versions per storypoint (may require some additional functionality in `sheet-to-triples`).
+- Once `ris3_points.ttl` can be recreated by running `ris3_views.py`, delete `ris3_points.ttl`.
+- Document the correct process for updating the RIS3 data model as a first step towards better documentation the map update process across all maps.
+- AL to do the tile generation himself at least once so that he understands how it works.
+- Version tile generation scripts.

```

Comment:
Not sure about this. They shouldn't exist - we need to be able to run this stuff in lambda, and that means the inputs must be comprehensive without requiring a human to add more parameters.

Reply:
Yes, as a long-term goal this makes sense. This is more of a short-to-medium term thing of "everything that goes into making a map should be versioned and documented" because I see that as the first step along the road of removing the box on the data ingest flow diagram labelled "Martin's Computer". Once it's versioned and has instructions it can be done on anyone's computer.